File: 35.search-insert-position/solution.kt
class Solution {
    fun searchInsert(nums: IntArray, target: Int): Int {
        var min = 0
        var max = nums.size
        var index = (min + max) / 2
        while (nums[index] != target) {
            if (nums[index] > target) {
                max = index
            } else {
                min = index + 1
            }
            if (min == max) return min
            index = (min + max) / 2
        }
        return index
    }
}

File: 35.search-insert-position/solution.erl
-spec search_insert(Nums :: [integer()], Target :: integer()) -> integer().
search_insert(Nums, Target) ->
  search_insert(Nums, Target, 0).
search_insert([], Target, Idx) ->
  Idx;
search_insert([Target | T], Target, Idx) ->
  Idx;
search_insert([H | T], Target, Idx) when H > Target ->
  Idx;
search_insert([H | T], Target, Idx) ->
  search_insert(T, Target, Idx + 1).
